### ISSUE TITLE
A grayed out GUI is kept gray until mouse is over it

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#1825]
**Describe the bug**
If on General Settings, the option "When the player interface is disabled, GUIs should" is marked as "Grey out all their controls", after the a moment of disabling the interface, the GUI won't un-grey back, and instead be kept grey (checkered) until mouse is over the GUI - forcing a redraw.

**AGS Version**
3.6.0.36 and master https://github.com/adventuregamestudio/ags/commit/2e6c0a8c07657fa3e54cdf7b16b8428ea8a70ccd

**Game**
The game below is just a Sierra Template game with an additional GUI with a button that has sprites, and an added `player.Say` dialog to cause it to grey once.
[TestGrayed.zip](https://github.com/adventuregamestudio/ags/files/9893996/TestGrayed.zip)

**To Reproduce**
1. In the above game, select the talking option of the sierra interface
2. Talk to yourself
3. Observe after the talk is over, the GUI is kept grey out

**Expected behavior**
The GUI should un-grey after the interface is no longer disabled.

**Screenshots**
![grey_bug_gui](https://user-images.githubusercontent.com/2244442/198840704-629980f4-d029-4b7d-ac33-72fcbdeda5e3.gif)

**Desktop:**
 - OS: Windows
 - Version 10

**Additional context**
[Originally reported on forums here](https://www.adventuregamestudio.co.uk/forums/ags-engine-editor-releases/ags-3-6-0-release-candidate-rc1/msg636650748/#msg636650748)

**Comment by DollarsQuest** [comment]
After doing some testing I think this triggers for me when a Display() function is the last blocking function called before handing back control to the player.
```
function Test1() {
  Display("This doesn't trigger the bug as a wait is called after the display.");
  Wait(40);
}

function Test2() {
  Wait(40);
  Display("This does trigger the bug.");
}

function Test3() {
  int someVariable = 0;
  Wait(40);
  Display("This also triggers the bug as below lines aren't blocking.");
  someVariable ++;
}
```

**Comment by ivan-mogilko** [comment]
The guis update on large changes, such as interface enable/disable, is triggered by a command `MarkAllGUIForUpdate()`. It's possible that this call is missing on certain change.

**Comment by ivan-mogilko** [comment]
Pushed a fix: 126a5e4c97581aabadbbd4410441b83f371e18ea

**Comment by ericoporto** [comment]
The fix, worked here! Thanks!
